### ISSUE TITLE
Enhancement: [no-shadow] <a short description of my proposal>

### ISSUE THREAD

**Issue by ximex** [typescript-eslint/typescript-eslint#6069]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-shadow/

### Description

I think no-shadow shouldn't show an error/warning if the variable name of the result of a higher order function (HOF) is the same as a local variable in the HOF as it is declared after the processing if the HOF. At least there should be an option to disable it.

### Fail

```typescript
const items = [1, 2, 3, 4];
const item = 1;
const _item = items.filter((item) => (item % 2));
```

### Pass

```typescript
const items = [1, 2, 3, 4];
const item = items.filter((item) => (item % 2));
```

**Comment by bradzacher** [comment]
Thanks for the issue!

This is an extension rule. That means we take the base rule and extend it to support TS syntax or features.

We don't break from the base rule to add new features unless they are specifically related to TS.

In this instance - this request is not related to TS, thus we shall not implement it. If you'd like to see it, feel free to raise a request with the ESLint team.

**Comment by ximex** [comment]
thanks for the info. will do
